decred/tests/unit/dcr/test_vsp_unit.py

```diff
@@ -0,0 +1,447 @@
+    bCopy[0] = 255
+    with pytest.raises(AssertionError):
+        vsp.PurchaseInfo.unblob(bCopy.bytes())
+
+    # too long
+    bCopy = encode.ByteArray(b, copy=True)
+    bCopy += b"\x00"
+    with pytest.raises(AssertionError):
+        vsp.PurchaseInfo.unblob(bCopy.bytes())
+
+
+poolStats = {
+    "AllMempoolTix": 12,
+    "APIVersionsSupported": [1, 2],
+    "BlockHeight": 368781,
+    "Difficulty": 88.50820708,
+    "Expired": 3,
+    "Immature": 0,
+    "Live": 28,
+    "Missed": 349,
+    "OwnMempoolTix": 0,
+    "PoolSize": 5759,
+    "ProportionLive": 0.004861955200555652,
+    "ProportionMissed": 0.3216589861751152,
+    "Revoked": 349,
+    "TotalSubsidy": 293.10719669,
+    "Voted": 736,
+    "Network": "testnet3",
+    "PoolEmail": "joe@dcrstakedinner.com",
+    "PoolFees": 0.5,
+    "PoolStatus": "Open",
+    "UserCount": 44,
+    "UserCountActive": 34,
+    "Version": "1.6.0-pre",
+}
+
+
+def test_pool_stats():
+
+    ps = vsp.PoolStats(poolStats)
+
+    assert ps.allMempoolTix == 12
+    assert ps.apiVersionsSupported == [1, 2]
+    assert ps.blockHeight == 368781
+    assert ps.difficulty == 88.50820708
+    assert ps.expired == 3
+    assert ps.immature == 0
+    assert ps.live == 28
+    assert ps.missed == 349
+    assert ps.ownMempoolTix == 0
+    assert ps.poolSize == 5759
+    assert ps.proportionLive == 0.004861955200555652
+    assert ps.proportionMissed == 0.3216589861751152
+    assert ps.revoked == 349
+    assert ps.totalSubsidy == 293.10719669
+    assert ps.voted == 736
+    assert ps.network == "testnet3"
+    assert ps.poolEmail == "joe@dcrstakedinner.com"
+    assert ps.poolFees == 0.5
+    assert ps.poolStatus == "Open"
+    assert ps.userCount == 44
+    assert ps.userCountActive == 34
+    assert ps.version == "1.6.0-pre"
+
+
+votingServiceProvider = {
+    "url": "https://www.dcrstakedinner.com",
+    "apiKey": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1Nzc0MzM0NDIsIm"
+    "lzcyI6Imh0dHBzOi8vd3d3LmRjcnN0YWtlZGlubmVyLmNvbSIsImxvZ2dlZEluQXMiOjQ2fQ."
+    "PEb000_TjQuBYxjRdh-VOaXMdV2GUw3_ZyIyp_tfpFE",
+    "netName": "testnet3",
+    "purchaseInfo": vsp.PurchaseInfo.parse(purchaseInfo),
+}
+
+
+def test_vsp_init():
+
+    pool = vsp.VotingServiceProvider(**votingServiceProvider)
+    assert pool.url == "https://www.dcrstakedinner.com"
+    assert pool.apiKey == votingServiceProvider["apiKey"]
+    assert pool.net.Name == "testnet3"
+    assert pool.purchaseInfo.ticketAddress == "Tcbvn2hiEAXBDwUPDLDG2SxF9iANMKhdVev"
+
+
+def test_vsp_blobbing():
+
+    pool = vsp.VotingServiceProvider(**votingServiceProvider)
+    b = vsp.VotingServiceProvider.blob(pool)
+    assert isinstance(b, bytearray)
+
+    rePool = vsp.VotingServiceProvider.unblob(b)
+    assert rePool.url == "https://www.dcrstakedinner.com"
+    assert rePool.apiKey == votingServiceProvider["apiKey"]
+    assert rePool.net.Name == "testnet3"
+    assert rePool.purchaseInfo.ticketAddress == "Tcbvn2hiEAXBDwUPDLDG2SxF9iANMKhdVev"
```

Comment:
Also can be a separate function, since you're doing this in `test_vsp_init` too. Can compare `rePool` directly to `pool`.